### PR TITLE
MODPUBSUB-35 - Create DAO components for module schema

### PR DESCRIPTION
**Pull request by RuslanLavrov**
[pub-sub design](https://github.com/TarasSpashchenko/rfcs/blob/rfc/pub-sub/text/0000-pub-sub.md)
pub-sub module schema diagram:
image

### CODE REVIEW

**Comment by OleksiiKuzminov** [comment]
Plz change id to uuid and add javadocs and descriptions for all fields

File: mod-pubsub-server/src/main/java/org/folio/dao/EventTypeDao.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * Event type data access object
+ */
+public interface EventTypeDao {
+
+  /**
+   * Searches for all {@link EventType} in database
+   *
+   * @return future with EventType list
+   */
+  Future<List<EventType>> getAll();
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/EventTypeDao.java]: filter(EventTypeFilter filter)

**Comment by RuslanLavrov** [comment]
> Plz change id to uuid and add javadocs and descriptions for all fields

changed module schema:
image

File: mod-pubsub-server/src/main/java/org/folio/dao/PostgresClientFactory.java [hunk 2]
```diff
@@ -20,4 +20,13 @@
   public PostgresClient getInstance(String tenantId) {
     return PostgresClient.getInstance(vertx, tenantId);
   }
+
+  /**
+   * Creates instance of Postgres Client
+   *
+   * @return Postgres Client
+   */
+  public PostgresClient createInstance() {
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/PostgresClientFactory.java]: getInstance

File: mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDescriptorDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      String query = format(UPDATE_BY_ID_SQL, MODULE_SCHEMA, TABLE_NAME);
+      JsonArray params = new JsonArray()
+        .add(pojo2json(eventDescriptor))
+        .add(eventDescriptor.getEventType());
+      pgClientFactory.createInstance().execute(query, params, future.completer());
+    } catch (Exception e) {
+      LOGGER.error("Error updating EventDescriptor by id '{}'", e, eventDescriptor.getEventType());
+      future.fail(e);
+    }
+    return future.compose(updateResult -> updateResult.getUpdated() == 1 ? Future.succeededFuture(eventDescriptor)
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDescriptorDaoImpl.java]: format code plz

Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDescriptorDaoImpl.java]: format code

**Pull request merged**